src/GeneratedInjectorDelegator.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-di for the canonical source repository
+ * @copyright Copyright (c) 2019 Zend Technologies USA Inc. (https://www.zend.com)
+ * @license   https://github.com/zendframework/zend-di/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Di;
+
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\ContainerInterface;
+use Zend\Di\Exception\UnexpectedValueException;
+use function class_exists;
+
+class GeneratedInjectorDelegator
+{
+    /**
+     * @param string $name
+     */
+    public function __invoke(ContainerInterface $container, $name, callable $callback) : InjectorInterface
+    {
+        $config = $container->has('config') ? $container->get('config') : [];
+        $aotConfig = $config['dependencies']['auto']['aot'] ?? [];
+        $namespace = empty($aotConfig['namespace']) ? 'Zend\Di\Generated' : $aotConfig['namespace'];
+
+        if (! is_string($namespace)) {
+            throw new class('Provided namespace is not a string.') extends UnexpectedValueException
+                implements ContainerExceptionInterface
```

Comment:
Not sure what exception should be thrown here. I think it should be instance of `ContainerExceptionInterface` as delegator is used in PSR-11 container.

Reply:
`ContainerExceptionInterface` is fine, but please make it a named class in the `Zend\Di\Exception` namespace. This will also solve the PHPUnit issue.